Updated tender import.

alexander-griffen opened the pull request:
This command pulls down _broadly_ correct and relevant data. That being said, I think that the better way to do it to get data that's more reliable is to pull down all (or most) of the data from the APIs.

Because this would take an extremely long time - at least initially - it's impractical to have this process happen with this command. It could be in an external repo that produces a csv that the import-tenders command could then ingest into the database. 

[I've written more about this, here.](https://docs.google.com/document/d/1d02ivH1lyzrtahR8R1BxC120ASvuz8OfHfvrJUcW5Jk/edit?usp=sharing)

alexander-griffen committed:
This command grabs data from the Contract Finder and Finder Tender APIs.

It takes about 25 minutes to run in full, but can also run updates based
on the last published date of a tender (though this isn't fully tested
in the case of tender updates over a longer period of time).

struan commented on procurement/management/commands/import_tenders.py:
```diff
@@ -21,46 +25,83 @@
 
     def handle(self, *args, **options):
         self.options = options
+        if self.check_valid_call():
```
You should be able to do this with the build in stuff for arg parse - https://docs.python.org/3/library/argparse.html#mutual-exclusion

struan commented on procurement/management/commands/import_tenders.py:
```diff
@@ -71,102 +112,207 @@ def get_groups(self):
+            self.council_lookup, left_on="buyer_name", right_on="la-name", how="inner"
+        )
+        df = df.query("tender_status == 'complete'").reset_index(drop=True)
+        # Reindex the df with all required ccolumns so that if they don't show up
```
ccolumns typo :)

struan commented on procurement/management/commands/import_tenders.py:
```diff
@@ -21,46 +25,83 @@
+    """
+    url = get_dataset_url(repo, package, version, file)
+    return pd.read_csv(url)
+
```
there's a `mysoc_dataset` package that has both these methods in it.

struan reviewed (approved):
Looks good and seems to work. Some tiny reuse comments but otherwise looks good.

Might be nice to add a `@unittest.skip("awaiting updates")` to the relevant test class just to cut down on test fails.

zarino reviewed (approved):
Worked fine for me! (Running from a fresh database – before I noticed there was a `--get_all` option that would have dropped all the existing tenders for me, d’oh!)

I noticed a couple of contracts have a "total cost" of "£nan", eg:

- http://localhost:8000/contract/ocds-b5fd17-30dab93f-50af-4b90-af6f-fc95cad1b3c4/
- http://localhost:8000/contract/ocds-b5fd17-cbcbcc7a-cb30-4785-b6ad-b24b8e845693/

That last one is also an example of a number of contracts that start in the future (5th Dec), which I thought we were hiding from view? Others include:

- http://localhost:8000/contract/ocds-b5fd17-c4f4aa47-bd0a-4ce3-853c-8ce5e2339eeb/
- http://localhost:8000/contract/ocds-b5fd17-d1b2a1f6-2316-4e5f-95db-0678ff87ce0f/

alexander-griffen committed:
Remove contracts that haven't started yet from view.

Also, change the tender info for values and start dates to come from the
award data, as that seems to be more accurate (and have fewer missing
values)
